src/maintainer/knowledge_base.rst

```diff
@@ -1145,6 +1145,45 @@ of your recipe and rerendering.
 
 Finally, note that the ``aarch64`` and ``ppc64le`` platforms already use CentOS 7.
 
+
+.. _opencl:
+
+OpenCL builds
+=============
+
+Including OpenCL can accelerate computation of supporting software without
+requiring packages to rebuild depending on the specific underlying hardware they are targetting.
+
+To build packages with OpenCL support, one must add the following to the relevant sections in the 
+recipe:
+
+.. code-block:: yaml
+
+    requirements:
+      build:
+        ...
+        # Include the opencl headers
+        # https://github.com/conda-forge/clhpp-feedstock
+        - clhpp
+        # Include the different loader implementations
+        - ocl-icd                    # [linux]
+        - khronos-opencl-icd-loader  # [osx or win]
+      run:
+        - ocl-icd-system         # [linux]
+        - ocl_icd_wrapper_apple  # [osx]
+        # We currently don't have support for running
+        # opencl applications on windows
+        # intel opencl implementation for windows
+
+    test:
+      requires:
+        - pocl  >=0.14      # [linux]
+
+If ``pocl`` is installed at the same time as ``ocl-icd-system`` or ``ocl_icd_wrapper_apple`` then
+``________`` will take precedence.
```

Comment:
There's no precedence. The application itself has to select one.

Reply:
So do we have a recommendation for which to "depend" on in the run section?

Reply:
I don't think you are understanding what I said about loaders. Here's a more detailed description https://documen.tician.de/pyopencl/misc.html#installing-from-conda-forge. Please read it entirely and let me know if you need clarification.

Reply:
hmm. maybe the software I was trying to build has a bug in it.

As I understand it:

It is up to the software itself to decide what implementation to use, if any are even found.

Therefore, the underlying application must:
1. Check if ANY opencl loaders are available. <-- my application was failing here.
2. Once it finds opencl loaders, the application itself must decide which to use.  <-- my application just chooses the first one that is returned.

Reply:
I changed `loaders` -> `implementations` and it's accurate.

For 2. applications like `pyopencl` asks the user what to do.

Reply:
as a final question. What should happen if absolutely no loader is found?

Reply:
I think the answer to that question is that the application is in charge of handling that event correctly.

Therefore, I think the correct suggestion is not to depend on anything in the run section.

Only in the test section if the packager wants to test the opencl codepath.

Reply:
Right, but you should add `ocl-icd-system` and `ocl_icd_wrapper_apple` which are not implementations, but just meta packages that enable the loader in conda-forge to see implementations in the system.

Reply:
I feel like system integrators should get the choice to allow OpenCL applications to find system implementations or not.

By default, I would prefer that packages not require system implementations to be available.